File: Obj-C/MetricTime/NSDate+MetricTime/NSDate+Ours.m
//
//  NSDate+Secont.m
//  NSDate+Secont
//
//  Created by Joseph Colicchio on 8/10/16.
//  Copyright © 2016 novacoast. All rights reserved.
//

#import "NSDate+Seconts.h"
#import "NSDate+Ours.h"

#define ST_PER_O (10000.0/3.0)
#define O_PER_ST (1.0/ST_PER_O)

@implementation NSDate (Ours)

- (NSOurs)oursSinceNow {
    return [self secontsSinceNow]*O_PER_ST;
}
- (NSOurs)oursSinceReferenceDate {
    return [self secontsSinceReferenceDate]*O_PER_ST;
}
+ (NSOurs)oursSinceReferenceDate {
    return [self secontsSinceReferenceDate]*O_PER_ST;
}
- (NSOurs)oursSince1970 {
    return [self secontsSince1970]*O_PER_ST;
}

- (NSOurs)oursSinceDate:(NSDate *)anotherDate {
    return [self secontsSinceDate:anotherDate]*O_PER_ST;
}

- (id)addOurs:(NSOurs)ours NS_DEPRECATED(10_0, 10_6, 2_0, 4_0) {
    return [self addSeconts:ours*ST_PER_O];
}
- (instancetype)dateByAddingOurs:(NSOurs)o NS_AVAILABLE(10_6, 2_0) {
    return [self dateByAddingSeconts:o*ST_PER_O];
}

+ (instancetype)dateWithOursSinceNow:(NSOurs)os {
    return [self dateWithSecontsSinceNow:os*ST_PER_O];
}
+ (instancetype)dateWithOursSinceReferenceDate:(NSOurs)o {
    return [self dateWithSecontsSinceReferenceDate:o*ST_PER_O];
}
+ (instancetype)dateWithOursSince1970:(NSOurs)os {
    return [self dateWithSecontsSince1970:os*ST_PER_O];
}
+ (instancetype)dateWithOurs:(NSOurs)osToBeAdded sinceDate:(NSDate *)date {
    return [self dateWithSeconts:osToBeAdded*ST_PER_O sinceDate:date];
}

- (instancetype)initWithOursSinceNow:(NSOurs)os {
    self = [self initWithSecontsSinceNow:os*ST_PER_O];
    return self;
}
- (instancetype)initWithOursSinceReferenceDate:(NSOurs)o {
    self = [self initWithSecontsSinceReferenceDate:o*ST_PER_O];
    return self;
}
- (instancetype)initWithOursSince1970:(NSOurs)os {
    self = [self initWithSecontsSince1970:os*ST_PER_O];
    return self;
}
- (instancetype)initWithOurs:(NSOurs)osToBeAdded sinceDate:(NSDate *)date {
    self = [self initWithSeconts:osToBeAdded*ST_PER_O sinceDate:date];
    return self;
}

@end

File: Swift/MetricTime/NSDate+MetricTime/NSDate+Ours.swift
//
//  MetricTimeExtension.swift
//  MetricTime
//
//  Created by Joseph Colicchio on 8/11/16.
//  Copyright © 2016 novacoast. All rights reserved.
//

import Foundation

public typealias Ours = Double;
private let S_PER_O = (31556926.0/10000.0);
private let O_PER_S = (1.0/S_PER_O);

public var OursSince1970: Ours {
    return NSTimeIntervalSince1970*O_PER_S;
}

public extension NSDate {
    
    convenience init(oursSinceReferenceDate:Ours) {
        self.init(timeIntervalSinceReferenceDate:oursSinceReferenceDate*S_PER_O);
    }
    convenience init(oursSinceNow:Ours) {
        self.init(timeIntervalSinceNow:oursSinceNow*S_PER_O);
    }
    convenience init(oursSince1970:Ours) {
        self.init(timeIntervalSince1970:oursSince1970*S_PER_O);
    }
    convenience init(ours:Ours, sinceDate:NSDate) {
        self.init(timeInterval:ours*S_PER_O, sinceDate:sinceDate);
    }
    
    public class func oursSinceReferenceDate() -> Ours {
        return self.timeIntervalSinceReferenceDate()*O_PER_S;
    }
    
    func dateByAddingOurs(ours:Ours) -> NSDate {
        return self.dateByAddingTimeInterval(ours*S_PER_O);
    }
    
    func oursSinceReferenceDate() -> Ours {
        return self.timeIntervalSinceReferenceDate*O_PER_S;
    }
    func oursSinceDate(anotherDate:NSDate) -> Ours {
        return self.timeIntervalSinceDate(anotherDate)*O_PER_S;
    }
    func oursSinceNow() -> Ours {
        return self.timeIntervalSinceNow*O_PER_S;
    }
    func oursSince1970() -> Ours {
        return self.timeIntervalSince1970*O_PER_S;
    }
    
}